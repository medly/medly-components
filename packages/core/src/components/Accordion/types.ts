export type StaticProps = {
    Header: React.FC;
    Content: React.FC;
    Context: React.Context<AccordionContextType>;
};

export type AccordionContextType = [
    boolean /** Current active state */,
    React.Dispatch<React.SetStateAction<boolean>> /** Function to be called to toggle the active state */
];
