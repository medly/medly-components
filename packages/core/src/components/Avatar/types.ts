import { AvatarSizes } from '@medly-components/theme';
import { HTMLProps, Omit } from '@medly-components/utils';

export interface AvatarProps extends Omit<HTMLProps<HTMLDivElement>, 'size'> {
    /** Size of the avatar */
    size?: AvatarSizes;
    /** Text color */
    textColor?: string;
    /** Background color */
    bgColor?: string;
    /** Hover Text color */
    hoverTextColor?: string;
    /** Hover Background color */
    hoverBgColor?: string;
    /** Hover text shadow color*/
    hoverTextShadowColor?: string;
    /** Hover image shadow color */
    hoverImgShadowColor?: string;
    /** Pressed Background color */
    pressedBgColor?: string;
    /** Pressed Text color */
    pressedTextColor?: string;
}

export interface StyledProps extends AvatarProps {
    /** To check children is image */
    isImage?: boolean;
}
