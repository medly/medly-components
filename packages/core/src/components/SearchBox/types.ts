import { Size } from '@medly-components/theme';
import { HTMLProps } from '@medly-components/utils';
import { ReactElement, ReactNode } from 'react';

export type Option = {
    label: string;
    value: string | number | JSX.Element;
    selected?: boolean;
};

export type WrapperProps = {
    size?: Size;
    isActive?: boolean;
    areOptionsVisible?: boolean;
};

export interface SearchBoxProps extends Omit<HTMLProps<HTMLInputElement>, 'size'> {
    /** Size for search box can be 'S' | 'M' */
    size?: Size;
    /** Option for search results in form of label and value */
    options?: Option[];
    /** React component for additional filtering */
    customSearchFilter?: ReactElement<any>;
    /** Show loader while the flag is true */
    isLoading?: boolean;
    /** Pass any React component as a loader */
    loader?: ReactNode;
    /** Does the component take up all available width */
    fullWidth?: boolean;
    /** Min width in rem/% (1rem = 10px)*/
    minWidth?: string;
    /** Max width in px/rem/% (1rem = 10px)*/
    maxWidth?: string;
    /** Function to be called on clear icon click */
    onClear?: () => void;
    /** Function to be called on search icon click */
    onSearch?: (value: string) => void;
    /** Function to be called on  selecting the option */
    onOptionSelected?: (value: Option) => void;
    /** Function to be called on input value changes */
    onInputChange?: (value: string) => void;
}

export type StyledSearchBoxProps = SearchBoxProps & {
    size: Size;
};
