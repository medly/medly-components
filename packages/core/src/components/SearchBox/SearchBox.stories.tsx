import { defaultTheme, SearchBoxTheme } from '@medly-components/theme';
import type { FC } from 'react';
import Button from '../Button';
import Text from '../Text';
import TextField from '../TextField';
import { SearchBoxProps } from './types';

export const PlaceholderComponent: FC = () => (
    <div>
        <TextField fullWidth variant="outlined" size="S" label="Organization" />
        <Button size="S">Submit</Button>
    </div>
);

export const optionsArray = [
    { value: 'Dummy1 option', label: 'Dummy1 option' },
    { value: 'Some 1', label: 'Dummy2 option' },
    { value: 'Dummy3 option', label: 'Dummy3 option' },
    { value: 'Dummy4 option', label: 'Dummy4 option' },
    { value: 'Dummy5 option', label: 'Dummy5 option' },
    { value: 'Dummy6 option', label: 'Dummy6 option' },
    { value: 'Dummy7 option', label: 'Dummy7 option' },
    { value: <Text>Custom component</Text>, label: 'Custom component' }
];

export const size: SearchBoxProps['size'][] = ['S', 'M'];

export const ThemeInterface: FC<SearchBoxTheme> = () => null;
ThemeInterface.defaultProps = {
    ...defaultTheme.searchBox
};
