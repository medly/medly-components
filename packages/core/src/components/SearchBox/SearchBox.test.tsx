import { fireEvent, render, screen } from '@test-utils';
import { SearchBox } from './SearchBox';
import { PlaceholderComponent } from './SearchBox.stories';
import { SearchBoxProps } from './types';

function renderComponent(props: SearchBoxProps) {
    const { container, ...rest } = render(<SearchBox {...props} />);

    return {
        container,
        inputEl: screen.getByPlaceholderText('search') as HTMLInputElement,
        ...rest
    };
}

describe('SearchBox', () => {
    const sizes: Required<SearchBoxProps>['size'][] = ['S', 'M'];

    test.each(sizes)('should render properly with %p size', size => {
        const { container } = render(<SearchBox size={size} />);
        expect(container).toMatchSnapshot();
    });

    it('should render placeholder text when specified', () => {
        const { inputEl } = renderComponent({ placeholder: 'search' });
        expect(inputEl).toBeInTheDocument();
    });

    it('should render search icon', () => {
        render(<SearchBox />);
        expect(screen.getByTitle('search icon')).toBeInTheDocument();
    });

    it('should call onSearch on clicking on search icon with input value', () => {
        const onSearch = jest.fn();
        const { inputEl } = renderComponent({ onSearch, onInputChange: jest.fn(), placeholder: 'search' });
        fireEvent.change(inputEl, { target: { value: 'R' } });
        fireEvent.click(screen.getByTitle('search icon'));
        expect(onSearch).toHaveBeenCalledWith('R');
    });

    it('should call onSearch on pressing enter key with the input value', () => {
        const onSearch = jest.fn(),
            withOptionCB = { placeholder: 'search', onInputChange: jest.fn(), onSearch };

        const { container, inputEl } = renderComponent(withOptionCB);
        fireEvent.focus(inputEl);
        fireEvent.change(inputEl, { target: { value: 'Dummy' } });
        fireEvent.keyDown(container, { key: 'Enter', code: 13 });
        expect(onSearch).toHaveBeenCalledWith('Dummy');
    });

    it('should call onClear on clicking on clear icon', () => {
        const onClear = jest.fn(),
            { container, inputEl } = renderComponent({ placeholder: 'search', onInputChange: jest.fn(), onClear });
        fireEvent.change(inputEl, { target: { value: 'Dummy' } });
        fireEvent.keyDown(container, { key: 'Enter', code: 13 });
        fireEvent.click(screen.getByTitle('close icon'));
        expect(onClear).toHaveBeenCalledWith();
        expect(inputEl.value).toEqual('');
    });

    describe('close icon', () => {
        const props = {
            placeholder: 'search',
            onInputChange: jest.fn()
        };

        it('should render close icon when user provides initial value', () => {
            const { inputEl } = renderComponent({ ...props, value: 'ABC' });
            expect(inputEl.value).toHaveLength(3);
            expect(screen.getByTitle('close icon')).toBeInTheDocument();
        });

        it('should render close icon when user starts typing', () => {
            const { inputEl } = renderComponent(props);
            fireEvent.change(inputEl, { target: { value: 'R' } });
            expect(inputEl.value).toHaveLength(1);
            expect(screen.getByTitle('close icon')).toBeInTheDocument();
        });

        it('should not render close icon in initial state', () => {
            renderComponent(props);
            expect(screen.queryByTitle('close icon')).not.toBeInTheDocument();
        });

        it('should hide the close icon and maintain the focus on input when clicked on close icon', () => {
            const { inputEl } = renderComponent(props);
            fireEvent.change(inputEl, { target: { value: 'R' } });
            const closeIcon = screen.getByTitle('close icon');
            expect(closeIcon).toBeInTheDocument();
            fireEvent.click(closeIcon);
            expect(inputEl.value).toHaveLength(0);
            expect(screen.queryByTitle('close icon')).not.toBeInTheDocument();
            expect(document.activeElement === inputEl).toBeTruthy();
        });

        it('should hide the close icon when length of input is 0', () => {
            const { inputEl } = renderComponent(props);
            fireEvent.change(inputEl, { target: { value: 'R' } });
            fireEvent.change(inputEl, { target: { value: '' } });
            expect(screen.queryByTitle('close icon')).not.toBeInTheDocument();
        });

        describe('when submitting a search with text', () => {
            it('should show the close icon after pressing enter', () => {
                const { container, inputEl } = renderComponent(props);
                fireEvent.change(inputEl, { target: { value: 'Brooklyn' } });
                const closeIcon = screen.getByTitle('close icon');
                expect(closeIcon).toBeInTheDocument();
                fireEvent.keyDown(container, { key: 'Enter', code: 13 });
                expect(inputEl.value).toHaveLength(8);
                expect(screen.getByTitle('close icon')).toBeInTheDocument();
            });

            it('should show the close icon after clicking the search icon', () => {
                const { inputEl } = renderComponent(props);
                fireEvent.change(inputEl, { target: { value: 'Brooklyn' } });
                const closeIcon = screen.getByTitle('close icon');
                expect(closeIcon).toBeInTheDocument();
                fireEvent.click(screen.getByTitle('search icon'));
                expect(inputEl.value).toHaveLength(8);
                expect(screen.getByTitle('close icon')).toBeInTheDocument();
            });
        });
    });

    describe('expand icon', () => {
        const props = {
            placeholder: 'search',
            customSearchFilter: <PlaceholderComponent />
        };

        it('should render expand icon when showExpandIcon prop is true', () => {
            renderComponent(props);
            expect(screen.getByTitle('expand icon')).toBeInTheDocument();
        });
    });

    describe('loading state', () => {
        it('should show loader when isLoading is true', () => {
            render(<SearchBox isLoading />);
            expect(screen.getByTestId('circle-loader')).toBeInTheDocument();
        });

        it('should not show search icon when isLoading is true', () => {
            render(<SearchBox isLoading />);
            expect(screen.queryByTitle('search icon')).not.toBeInTheDocument();
        });

        it('should show custom loader when passed', () => {
            const CustomLoader = () => <span>Loading</span>;
            render(<SearchBox loader={<CustomLoader />} isLoading />);
            expect(screen.getByText('Loading')).toBeInTheDocument();
        });
    });

    describe('options', () => {
        const defaultReturnObj = { target: { value: 'Dummy' } };
        const props = {
            placeholder: 'search',
            onInputChange: jest.fn(),
            options: [
                { label: 'Dummy 1', value: 'Dummy 1' },
                { label: 'Dummy 2', value: 'Dummy 2' }
            ]
        };

        it('should not show options initially', () => {
            renderComponent(props);
            expect(screen.queryByRole('list')).not.toBeInTheDocument();
        });

        it('should hide options when clicked outside', () => {
            render(
                <>
                    <SearchBox {...props} />
                    <span>Outside span</span>
                </>
            );
            const inputEl = screen.getByPlaceholderText('search') as HTMLInputElement;
            fireEvent.change(inputEl, defaultReturnObj);
            expect(screen.getByRole('list').children).toHaveLength(2);
            fireEvent.click(screen.getByText('Outside span'));
            expect(screen.queryByRole('list')).not.toBeInTheDocument();
        });

        it('should render options when user search option specified', () => {
            const { container, inputEl } = renderComponent({ ...props, size: 'S' });
            fireEvent.change(inputEl, defaultReturnObj);
            expect(screen.getByRole('list').children).toHaveLength(2);
            expect(container).toMatchSnapshot();
        });

        it('should render options on arrow keys when input is in focus', () => {
            const { inputEl } = renderComponent({ ...props, size: 'M' });
            fireEvent.focus(inputEl, defaultReturnObj);
            fireEvent.keyDown(inputEl, { key: 'ArrowDown', code: 40 });
            expect(screen.getByRole('list').children).toHaveLength(2);
        });

        it('should not render options on arrow keys when input is not in focus', () => {
            const { inputEl } = renderComponent(props);
            fireEvent.blur(inputEl, defaultReturnObj);
            fireEvent.keyDown(inputEl, { key: 'ArrowDown', code: 40 });
            expect(screen.queryByRole('list')).not.toBeInTheDocument();
        });

        it('should not call on OptionSelected when onOptionSelected is not passed', () => {
            const { container, inputEl } = renderComponent(props);
            fireEvent.change(inputEl, defaultReturnObj);
            fireEvent.keyDown(inputEl, { key: 'ArrowDown', code: 40 });
            fireEvent.keyDown(container, { key: 'Enter', code: 13 });

            expect(screen.getByRole('list')).toBeInTheDocument();
        });

        describe('with option selected', () => {
            const onOptionSelectedMock = jest.fn(),
                onSearch = jest.fn(),
                withOptionCB = { ...props, onOptionSelected: onOptionSelectedMock, onSearch };

            it('should call on OptionSelected when option is clicked', () => {
                const { inputEl } = renderComponent(withOptionCB);
                fireEvent.change(inputEl, defaultReturnObj);
                fireEvent.click(screen.getByText('Dummy 2'));
                expect(onOptionSelectedMock).toHaveBeenCalledWith({ value: 'Dummy 2', label: 'Dummy 2' });
            });

            it('should update the searchbox value to selected option', () => {
                const { container, inputEl } = renderComponent(withOptionCB);
                fireEvent.focus(inputEl);
                fireEvent.keyDown(container, { key: 'ArrowDown', code: 40 });
                fireEvent.keyDown(container, { key: 'Enter', code: 13 });
                expect(inputEl).toHaveValue('Dummy 1');
            });

            it('should not call onSearch on option selection using enter', () => {
                const { container, inputEl } = renderComponent(withOptionCB);
                fireEvent.focus(inputEl);
                fireEvent.keyDown(container, { key: 'ArrowDown', code: 40 });
                fireEvent.keyDown(container, { key: 'Enter', code: 13 });
                expect(onSearch).not.toBeCalled();
                expect(inputEl).toHaveValue('Dummy 1');
            });
        });
    });
});
