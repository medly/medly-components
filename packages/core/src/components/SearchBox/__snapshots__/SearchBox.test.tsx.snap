// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox options should render options when user search option specified 1`] = `
.c4 {
  overflow: visible;
  font-size: 2rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: pointer;
}

.c4 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c12 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
  display: inline-block;
}

.c10 {
  cursor: pointer;
  padding: 0 0.8rem;
  border-radius: 0.4rem;
  box-sizing: border-box;
  min-height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  background-color: #ffffff;
  max-width: 100%;
}

.c10:hover {
  background-color: #F7F8F9;
}

.c10:hover > .c11 {
  color: #13181D;
}

.c10:active {
  background-color: #eff2f4;
}

.c10 > .c11 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  white-space: nowrap;
  color: #13181D;
}

.c10 > .c3 {
  margin-left: 1.6rem;
}

.c10 > .c3 * {
  fill: #13181D;
}

.c10 > .c11 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

.c8 {
  z-index: 4;
  position: absolute;
  background: #ffffff;
  top: 4rem;
  left: 0;
  margin: 0;
  padding: 0.8rem;
  width: 100%;
  list-style: none;
  box-sizing: border-box;
  max-width: 100%;
  box-shadow: 0 0.2rem 0.8rem #B0BCC8;
  border-radius: 0 0 0.4rem 0.4rem;
  overflow: auto;
  max-height: 20rem;
}

.c2 {
  width: 4rem;
  min-width: 4rem;
  height: auto;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-right: 0.1rem solid #dfe4e9;
  margin: 0.6rem 0;
}

.c2 .c3 {
  padding: 0.4rem;
  border-radius: 2.5rem;
}

.c2 .c3 * {
  fill: #546A7F;
}

.c2 .c3:hover {
  background: #eff2f4;
}

.c2 .c3:hover * {
  fill: #435465;
}

.c2 .c3:active {
  background: #dfe4e9;
}

.c2 .c3:active * {
  fill: #303C48;
}

.c14 .c3 {
  padding: 0.4rem;
  border-radius: 2.5rem;
}

.c14 .c3 * {
  fill: #546A7F;
}

.c14 .c3:hover {
  background: #EBF1FA;
}

.c14 .c3:active {
  background: #D7E3F6;
}

.c14 .c3:active * {
  fill: #275093;
}

.c6 {
  width: 4rem;
  min-width: 4rem;
  height: auto;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 .c3 {
  padding: 0.4rem;
  border-radius: 2.5rem;
}

.c6 .c3 * {
  fill: #3872D2;
}

.c6 .c3:hover {
  background: #EBF1FA;
}

.c6 .c3:active {
  background: #D7E3F6;
}

.c6 .c3:active * {
  fill: #275093;
}

.c13 .c7 {
  box-shadow: none;
  border: none;
  padding: 0;
  box-sizing: content-box;
  left: -1.5px;
  width: calc(100% + 0.3rem);
  overflow-x: hidden;
  overflow-y: auto;
  background-color: #ffffff;
  border: none;
  padding-bottom: 0;
  max-height: 21rem;
  border-radius: 0.8rem;
  top: 3.9rem;
}

.c13 .c7 svg {
  display: none;
}

.c13 .c7 strong {
  font-weight: normal;
}

.c13 .c7 .c9:hover {
  background-color: #F7F8F9;
}

.c13 .c7 li {
  width: calc(100% - 0.8rem);
  margin: 0 0.4rem;
}

.c13:focus-within .c5 .c3 * {
  fill: #3872D2;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  outline: none;
  box-sizing: border-box;
  height: 4rem;
  border: 1.5px solid;
  position: relative;
  padding: 0 0 0 1.2rem;
  background-color: #ffffff;
  min-width: 25.6rem;
  max-width: 100%;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  border-radius: .8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(18,106,250,0.2);
  border-color: #3872D2;
  border-width: 0.2rem;
}

.c0 input {
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
}

.c0 .c7 {
  box-shadow: none;
  border: none;
  padding: 0;
  box-sizing: content-box;
  left: -1.5px;
  width: calc(100% + 0.3rem);
  overflow-x: hidden;
  overflow-y: auto;
  background-color: #ffffff;
  border-radius: 0.8rem;
  box-shadow: 0 0.2rem 0.8rem rgba(96,120,144,0.35);
  padding: 0.4rem 0;
  max-height: 21rem;
  border-radius: 0.8rem;
  top: 3.9rem;
}

.c0 .c7 svg {
  display: none;
}

.c0 .c7 strong {
  font-weight: normal;
}

.c0 .c7 .c9:hover {
  background-color: #F7F8F9;
}

.c0 .c7 li {
  width: calc(100% - 0.8rem);
  margin: 0 0.4rem;
}

.c1 {
  background: transparent;
  color: #13181D;
  padding: 0;
  outline: none;
  border: 0 none;
  min-width: 12.9rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-family: inherit;
  font-weight: 600;
}

.c1::-webkit-input-placeholder {
  color: #607890;
}

.c1::-moz-placeholder {
  color: #607890;
}

.c1:-ms-input-placeholder {
  color: #607890;
}

.c1::placeholder {
  color: #607890;
}

.c1:focus::-webkit-input-placeholder {
  color: #C7D0D8;
}

.c1:focus::-moz-placeholder {
  color: #C7D0D8;
}

.c1:focus:-ms-input-placeholder {
  color: #C7D0D8;
}

.c1:focus::placeholder {
  color: #C7D0D8;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      placeholder="search"
      value=""
    />
    <span
      class="c2"
    >
      <svg
        class="c3 c4"
        fill="none"
        height="1em"
        title="close icon"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M18.3 5.71a.996.996 0 0 0-1.41 0L12 10.59 7.11 5.7A.996.996 0 1 0 5.7 7.11L10.59 12 5.7 16.89a.996.996 0 1 0 1.41 1.41L12 13.41l4.89 4.89a.996.996 0 1 0 1.41-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
          fill="#000"
          fill-opacity=".54"
        />
      </svg>
    </span>
    <span
      class="c5 c6"
    >
      <svg
        class="c3 c4"
        fill="none"
        height="1em"
        title="search icon"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M9.063 3.125a5.938 5.938 0 1 0 0 11.875 5.938 5.938 0 0 0 0-11.875zM1.875 9.063a7.187 7.187 0 1 1 12.692 4.621l3.375 3.374a.625.625 0 1 1-.884.884l-3.374-3.374A7.188 7.188 0 0 1 1.875 9.063z"
          fill="#606878"
          fill-rule="evenodd"
          stroke="#606878"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width=".3"
        />
      </svg>
    </span>
    <ul
      class="c7 c8"
    >
      <li
        class="c9 c10"
        id="Dummy-1"
      >
        <span
          class="c11 c12"
        >
          Dummy 1
        </span>
      </li>
      <li
        class="c9 c10"
        id="Dummy-2"
      >
        <span
          class="c11 c12"
        >
          Dummy 2
        </span>
      </li>
    </ul>
  </div>
</div>
`;

exports[`SearchBox should render properly with "M" size 1`] = `
.c5 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: pointer;
}

.c5 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c3 {
  width: 4.8rem;
  min-width: 4.8rem;
  height: auto;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 .c4 {
  padding: 0.4rem;
  border-radius: 2.5rem;
}

.c3 .c4 * {
  fill: #546A7F;
}

.c3 .c4:hover {
  background: #EBF1FA;
}

.c3 .c4:active {
  background: #D7E3F6;
}

.c3 .c4:active * {
  fill: #275093;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  outline: none;
  box-sizing: border-box;
  height: 4.8rem;
  border: 1.5px solid;
  position: relative;
  padding: 0 0 0 1.6rem;
  background-color: #ffffff;
  min-width: 25.6rem;
  max-width: 100%;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  border-radius: .8rem;
  border-color: #98A7B7;
  -webkit-transition: border-color 100ms,box-shadow 100ms;
  transition: border-color 100ms,box-shadow 100ms;
}

.c0 input {
  font-size: 1.6rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.6rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
}

.c0:hover {
  background-color: #ffffff;
}

.c0 .sc-1kflsvi-0 {
  box-shadow: none;
  border: none;
  padding: 0;
  box-sizing: content-box;
  left: -1.5px;
  width: calc(100% + 0.3rem);
  overflow-x: hidden;
  overflow-y: auto;
  background-color: #ffffff;
  border: none;
  padding-bottom: 0;
  max-height: 21rem;
  border-radius: 0.8rem;
  top: 4.7rem;
}

.c0 .sc-1kflsvi-0 svg {
  display: none;
}

.c0 .sc-1kflsvi-0 strong {
  font-weight: normal;
}

.c0 .sc-1kflsvi-0 .sc-1yc1o3f-0:hover {
  background-color: #F7F8F9;
}

.c0 .sc-1kflsvi-0 li {
  width: calc(100% - 1.6rem);
  margin: 0 0.8rem;
}

.c0:focus-within {
  border-color: #3872D2;
  box-shadow: 0 0.2rem 0.8rem rgba(18,106,250,0.2);
  border-width: .2rem;
}

.c0:focus-within .c2 .c4 * {
  fill: #3872D2;
}

.c0:hover {
  border-color: #607890;
  box-shadow: 0 0.2rem 0.8rem rgba(96,120,144,0.2);
  border-width: 1.5px;
}

.c0:focus-within:hover {
  border-color: #3872D2;
  box-shadow: 0 0.2rem 0.8rem rgba(18,106,250,0.2);
  border-width: .2rem;
}

.c1 {
  background: transparent;
  color: #13181D;
  padding: 0;
  outline: none;
  border: 0 none;
  min-width: 12.9rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-family: inherit;
  font-weight: 600;
}

.c1::-webkit-input-placeholder {
  color: #607890;
}

.c1::-moz-placeholder {
  color: #607890;
}

.c1:-ms-input-placeholder {
  color: #607890;
}

.c1::placeholder {
  color: #607890;
}

.c1:focus::-webkit-input-placeholder {
  color: #C7D0D8;
}

.c1:focus::-moz-placeholder {
  color: #C7D0D8;
}

.c1:focus:-ms-input-placeholder {
  color: #C7D0D8;
}

.c1:focus::placeholder {
  color: #C7D0D8;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      placeholder="Search"
      value=""
    />
    <span
      class="c2 c3"
    >
      <svg
        class="c4 c5"
        fill="none"
        height="1em"
        title="search icon"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M9.063 3.125a5.938 5.938 0 1 0 0 11.875 5.938 5.938 0 0 0 0-11.875zM1.875 9.063a7.187 7.187 0 1 1 12.692 4.621l3.375 3.374a.625.625 0 1 1-.884.884l-3.374-3.374A7.188 7.188 0 0 1 1.875 9.063z"
          fill="#606878"
          fill-rule="evenodd"
          stroke="#606878"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width=".3"
        />
      </svg>
    </span>
  </div>
</div>
`;

exports[`SearchBox should render properly with "S" size 1`] = `
.c5 {
  overflow: visible;
  font-size: 2rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: pointer;
}

.c5 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c3 {
  width: 4rem;
  min-width: 4rem;
  height: auto;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 .c4 {
  padding: 0.4rem;
  border-radius: 2.5rem;
}

.c3 .c4 * {
  fill: #546A7F;
}

.c3 .c4:hover {
  background: #EBF1FA;
}

.c3 .c4:active {
  background: #D7E3F6;
}

.c3 .c4:active * {
  fill: #275093;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  outline: none;
  box-sizing: border-box;
  height: 4rem;
  border: 1.5px solid;
  position: relative;
  padding: 0 0 0 1.2rem;
  background-color: #ffffff;
  min-width: 25.6rem;
  max-width: 100%;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  border-radius: .8rem;
  border-color: #98A7B7;
  -webkit-transition: border-color 100ms,box-shadow 100ms;
  transition: border-color 100ms,box-shadow 100ms;
}

.c0 input {
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  font-family: Open Sans,Helvetica Neue,Helvetica,Arial,sans-serif;
}

.c0:hover {
  background-color: #ffffff;
}

.c0 .sc-1kflsvi-0 {
  box-shadow: none;
  border: none;
  padding: 0;
  box-sizing: content-box;
  left: -1.5px;
  width: calc(100% + 0.3rem);
  overflow-x: hidden;
  overflow-y: auto;
  background-color: #ffffff;
  border: none;
  padding-bottom: 0;
  max-height: 21rem;
  border-radius: 0.8rem;
  top: 3.9rem;
}

.c0 .sc-1kflsvi-0 svg {
  display: none;
}

.c0 .sc-1kflsvi-0 strong {
  font-weight: normal;
}

.c0 .sc-1kflsvi-0 .sc-1yc1o3f-0:hover {
  background-color: #F7F8F9;
}

.c0 .sc-1kflsvi-0 li {
  width: calc(100% - 0.8rem);
  margin: 0 0.4rem;
}

.c0:focus-within {
  border-color: #3872D2;
  box-shadow: 0 0.2rem 0.8rem rgba(18,106,250,0.2);
  border-width: .2rem;
}

.c0:focus-within .c2 .c4 * {
  fill: #3872D2;
}

.c0:hover {
  border-color: #607890;
  box-shadow: 0 0.2rem 0.8rem rgba(96,120,144,0.2);
  border-width: 1.5px;
}

.c0:focus-within:hover {
  border-color: #3872D2;
  box-shadow: 0 0.2rem 0.8rem rgba(18,106,250,0.2);
  border-width: .2rem;
}

.c1 {
  background: transparent;
  color: #13181D;
  padding: 0;
  outline: none;
  border: 0 none;
  min-width: 12.9rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-family: inherit;
  font-weight: 600;
}

.c1::-webkit-input-placeholder {
  color: #607890;
}

.c1::-moz-placeholder {
  color: #607890;
}

.c1:-ms-input-placeholder {
  color: #607890;
}

.c1::placeholder {
  color: #607890;
}

.c1:focus::-webkit-input-placeholder {
  color: #C7D0D8;
}

.c1:focus::-moz-placeholder {
  color: #C7D0D8;
}

.c1:focus:-ms-input-placeholder {
  color: #C7D0D8;
}

.c1:focus::placeholder {
  color: #C7D0D8;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      placeholder="Search"
      value=""
    />
    <span
      class="c2 c3"
    >
      <svg
        class="c4 c5"
        fill="none"
        height="1em"
        title="search icon"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M9.063 3.125a5.938 5.938 0 1 0 0 11.875 5.938 5.938 0 0 0 0-11.875zM1.875 9.063a7.187 7.187 0 1 1 12.692 4.621l3.375 3.374a.625.625 0 1 1-.884.884l-3.374-3.374A7.188 7.188 0 0 1 1.875 9.063z"
          fill="#606878"
          fill-rule="evenodd"
          stroke="#606878"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width=".3"
        />
      </svg>
    </span>
  </div>
</div>
`;
